src/modules/app/containers/Switcher.js

```diff
@@ -149,7 +211,6 @@ export class Switcher extends React.Component<PropsType> {
 }
 
 const mapStateToProps = (state: StateType) => ({
```

Comment:
Puh, that Switcher is still way too complicated imo. It depends on the complete state of the whole app.

Maybe we can solve this in the router upgrade. imo a router should handle the switching

Reply:
yeah I'll have a look at it again, maybe I can split it in several components. But I don't see how a different router could make a difference.

Reply:
@maxammann I think it's a lot better now, let me know what you think or if you have any suggestions how to make it simpler.